src/platform/i386/chal.c

```diff
@@ -89,14 +90,26 @@ chal_init(void)
 	if (d & (1 << 22)) printk("acpi ");
 	if (d & (1 << 9)) printk("apic ");
 	if (c & (1 << 21)) printk("x2apic ");
+	if (c & (1 << 24)) printk("tsc-deadline ");
 	chal_cpuid(0x80000001, &a, &b, &c, &d);
 	if (d & (1 << 27)) printk("rdtscp ");
 	chal_cpuid(0x80000007, &a, &b, &c, &d);
 	if (d & (1 << 8)) printk("invariant_tsc ");
 	printk("]\n");
 
+	chal_cpuid(0x16, &a, &b, &c, &d);
```

Comment:
Does this work on processors without the frequency support?  Do we not need a `if (cpuid_hz_support_check) {...}`?  If this works, on all processors, great!